### DISCUSSION TITLE
Arduino as ESP-IDF Component in Micropython Firmware

### DISCUSSION THREAD

**Discussion by BrunoESP32** [micropython/micropython#13452] [ESP32]
Hey,

I have many useful libraries for ESP32 on Arduino (c++) and would be happy if they are available in cppmodules for use in python. I was able to compile Micropython 1.20.0 with Arduino 2.0.0 [(Arduino 2.0.0 RC2)](https://github.com/espressif/arduino-esp32/releases/tag/2.0.0-rc2) (I tried other versions too) as a ESP-IDF 4.4 component and I managed to use some libraries, but most of them had a lot of problems or didn't work properly (those librieries that use WiFi always crash the esp32). I almost rewrote the entire sdkconfig looking for what could be preventing them from starting correctly, I even disabled MICROPY_BOARD_STARTUP() and mp_task, but the result was the same. Outside the Micropython 1.20.0 project, this Arduino component works correctly, but inside the project the story changes. 
Does anyone have any tips or know what could be preventing the proper functioning of the C++ libraries?

**Comment by jimmo** [comment]
hi @BrunoESP32 Sorry I'm not quite sure I understand the question... (possibly due to lack of familiarity with Arduino on ESP32).

Is your goal to be able to use the Arduino libraries from Python?

> compile Micropython 1.20.0 with Arduino 2.0.0 [(Arduino 2.0.0 RC2)](https://github.com/espressif/arduino-esp32/releases/tag/2.0.0-rc2)

This sounds difficult... the way we use the IDF kind of assumes that MicroPython is the "outer" build system.

Are you able to use these libraries in a regular IDF application? i.e. can you add them to one of the IDF sample apps?

Also, if possible I'd recommend using the latest MicroPython version because the IDF integration has changed significantly over the past few releases.
